### PR TITLE
ILL powder few fixes

### LINKED ISSUE #24851: LoadILLDiffraction : overwrite the sample log, if duplicated
LoadILLDiffraction to use the second log with the same name, if the entry is duplicated.

### PR DESCRIPTION
**Pull request by gvardany**
**Description of work.**

Fixes a couple of problems with ILL powder loader and reducer.

**To test:**

- [  ] 
```
PowderDiffILLDetScanReduction(Run='508093:508095', Output1D = True, OutputWorkspace='list')
PowderDiffILLDetScanReduction(Run='508093-508095', Output1D = True, OutputWorkspace='sum')
comp = CompareWorkspaces(Workspace1='list_1D', Workspace2='sum_1D')
assert(comp.Result)
```

- [  ]
```
PowderDiffILLDetEffCorr(CalibrationRun='967076.nxs', OutputWorkspace='calib', OutputResponseWorkspace='response')
```
   Plot `response`, make sure that the scattering angle 0 is within the beam stop range (the range with almost no intensity).

Data is in system tests under ILL/D2B and ILL/D20 respectively.

The rest are captured by the tests.

Fixes #24851 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/test/LoadILLDiffractionTest.h [hunk 1]
```diff
@@ -230,6 +231,29 @@ class LoadILLDiffractionTest : public CxxTest::TestSuite {
...
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("OutputWorkspace", "_outWS"))
+    TS_ASSERT_THROWS_NOTHING(alg.execute())
+    TS_ASSERT(alg.isExecuted())
+
+    MatrixWorkspace_sptr outputWS =
+        AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("_outWS");
+    TS_ASSERT(outputWS)
+
+    const auto detectorInfo = outputWS->detectorInfo();
+    const auto &indexOfFirstDet = detectorInfo.indexOf(1);
```
Comment (soininen) [inline Framework/DataHandling/test/LoadILLDiffractionTest.h]: The `&` is unnecessary as `indexOf()` returns `size_t`.

File: Framework/DataHandling/inc/MantidDataHandling/LoadILLDiffraction.h [hunk 2]
```diff
@@ -92,8 +92,8 @@ class MANTID_DATAHANDLING_DLL LoadILLDiffraction
   void setSampleLogs();
   void computeThetaOffset();
 
-  double
-      m_offsetTheta; ///< the 2theta offset for D20 to account for dead pixels
+  double m_offsetTheta{
+      0.};           ///< the 2theta offset for D20 to account for dead pixels
```
Comment (soininen) [inline Framework/DataHandling/inc/MantidDataHandling/LoadILLDiffraction.h]: The doxygen comment could be on the preceding line to get this formatted correctly.

File: Framework/API/inc/MantidAPI/LogManager.h [hunk 3]
```diff
@@ -121,13 +121,16 @@ class MANTID_API_DLL LogManager {
...
   }
+
   /**
    * Add a log entry
    * @param p :: A pointer to the property containing the log entry
+   * @param overwrite :: Overwrite existing if requested
    */
-  void addLogData(std::unique_ptr<Kernel::Property> p) {
-    addProperty(std::move(p));
+  void addLogData(std::unique_ptr<Kernel::Property> p, bool overwrite = false) {
```
Comment (soininen) [inline Framework/API/inc/MantidAPI/LogManager.h]: :+1:

File: Framework/DataHandling/test/LoadILLDiffractionTest.h [hunk 4]
```diff
@@ -230,6 +231,29 @@ class LoadILLDiffractionTest : public CxxTest::TestSuite {
...
+    const V3D position =
+        detectorInfo.position(std::make_pair(indexOfFirstDet, 0));
+    double r, theta, phi;
+    position.getSpherical(r, theta, phi);
+    TS_ASSERT_DELTA(theta, 5.825, 0.001);
+    TS_ASSERT(position.X() < 0.);
```
Comment (soininen) [inline Framework/DataHandling/test/LoadILLDiffractionTest.h]: `TS_ASSERT_LESS_THAN`?

File: docs/source/release/v3.14.0/diffraction.rst [hunk 5]
```diff
@@ -30,6 +30,8 @@ Improvements
   (see :ref:`ConvertToMD <algm-ConvertToMD>` Notes).
 - New TOPAZ instrument geometry for 2019 run cycle
 - :ref:`LoadDiffCal <algm-LoadDiffCal>` has an additional parameter to allow for a second file specifying a grouping to override the one in the calibration file
+- :ref:`LoadILLDiffraction <algm-LoadILLDiffraction>` will now correctly resolve for the scan type and drive the detector to the offset corrected 2theta0 for D20 detector scans.
```
Comment (soininen) [inline docs/source/release/v3.14.0/diffraction.rst]: `:math:'2theta_0'`?

**Review by soininen — CHANGES_REQUESTED** [review]
Looks good to me. Just a few tiny issues below.

**Review by soininen — APPROVED** [review]

**Comment by NickDraper** [comment]
Conflicts - A Haiku

```
To resolve conflicts
developer must merge
in the master branch
```

**Comment by soininen** [comment]
The failed test seems unrelated.

**Pull request merged**